### ISSUE TITLE
Fully qualified base types are not propagated in lcm-derive

### ISSUE THREAD

**Issue by neachdainn** [adeschamps/lcm-rust#5]
If a struct has a field that uses fully qualified syntax, then only the final part is output in the generated code. For example:

```rust
mod pixel {
    #[derive(LcmMessage)]
    pub struct Pixel {
        blue: u8,
        green: u8,
        red: u8,
    }
}

#[derive(LcmMessage)]
struct Image {
    data: [pixel::Pixel; 144];
}
```

produces code that references `Pixel` rather than `pixel::Pixel`.

This may also cause issues if people use a type called `Vec` that is not an `std::vec::Vec` or if they specify `std::vec::Vec`.

**Comment by neachdainn** [comment]
This same issue causes generics to lose the generic information. I.e, `field: UserType<i32>` is referenced as `UserType` instead.

**Comment by neachdainn** [comment]
Fixed in #9
